variables.tf

```diff
@@ -129,6 +198,20 @@ variable "public_network_access_enabled" {
   description = "Whether public network access is allowed for the Cognitive Account. Defaults to `false`."
 }
 
+variable "storage" {
+  type = object({
```

Comment:
According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.56.0/internal/services/cognitive/cognitive_account_resource.go#L274-L277), it looks like the type here should be a list: `list(object({ ... }))`